llvm/lib/CodeGen/LiveDebugVariables.cpp

```diff
@@ -1341,7 +1339,7 @@ bool LiveDebugVariables::invalidate(
   // Some architectures split the register allocation into multiple phases based
   // on register classes. This requires preserving analyses between the phases
   // by default.
-  return PAC.preservedWhenStateless();
+  return !PAC.preservedWhenStateless();
```

Comment:
All passes are preserving this analysis and VirtRegRewriter will abandon this if it's the final one in the pipeline (`ClearVirtRegs = true`)